platform/core/src/DICOMSR/dataExchange.js

```diff
@@ -119,7 +80,11 @@ const generateReport = measurementData => {
  * that you wish to serialize.
  * @param {object} dataSource The dataSource that you wish to use to persist the data.
  */
-const storeMeasurements = async (measurementData, dataSource) => {
+const storeMeasurements = async (
+  measurementData,
+  dataSource,
+  displaySetService
```

Comment:
This method feels weird to me. DICOMSR/dataExchange, but it's coordinating behavior between a `WebApiDataSource` and has a sideEffect on `displaySetService`?

Where else is this used? It's already async/promise. I think we should either return something more useful, await that result, and then do follow-up stuff: notification, update displaySetService, etc.

Or `displaySetService` parameter should be swapped out for an onSuccess callback that receives that naturalizedReport

Reply:
I agree with the onSuccess, I guess you wouldn't always want to make displaySets locally 👍

Reply:
The `deleteStudyMetadataPromise` should be part of the main function call though, otherwise you will get caching issues as you _will_ have new data on your server. Its also like this on main branch.

Reply:
Updated.